yellowbrick/classifier/threshold.py

```diff
@@ -0,0 +1,351 @@
+                random_state=self.random_state)
+            self.estimator.fit(train_X, train_y)
+            # get prediction probabilities for each
+            predictions = self.estimator.predict_proba(test_X)[:, 1]
+
+            precision, recall, thresholds = precision_recall_curve(
+                test_y, predictions)
+            # add one to each so that thresh ends at 1
+            thresholds = np.append(thresholds, 1)
+            queue_rate = []
+            for threshold in thresholds:
+                queue_rate.append((predictions >= threshold).mean())
+
+            trial_data = {
+                'thresholds': thresholds,
+                'precision': precision,
+                'recall': recall,
+                'queue_rate': queue_rate
+            }
+            self.plot_data.append(trial_data)
+
+        return self
+
+    def draw(self, *kwargs):
+        """
+        Renders the visualization
+
+        Parameters
+        ----------
+        kwargs: dict
+            keyword arguments passed to Scikit-Learn API.
+
+        Returns
+        -------
+        self.ax : AxesSubplot of the visualizer
+            Returns the AxesSubplot instance of the visualizer
+        """
+        # set the colors
+        if self.colormap is not None or self.color is not None:
+            color_values = resolve_colors(
+                num_colors=3, colormap=self.colormap, color=self.color)
+        else:
+            color_values = get_color_cycle()
+
+        # unpack first three colors for visualization
+        recall_color, precision_color, queue_rate_color = color_values[:3]
+
+        if self.ax is None:
+            self.ax = plt.gca(xlim=[-1, 1], ylim=[-1, 1])
+
+        uniform_thresholds = np.linspace(0, 1, num=101)
+        uniform_precision_plots = []
+        uniform_recall_plots = []
+        uniform_queue_rate_plots = []
+
+        for data in self.plot_data:
+            uniform_precision = []
+            uniform_recall = []
+            uniform_queue_rate = []
+            for ut in uniform_thresholds:
+                index = bisect.bisect_left(data['thresholds'], ut)
+                uniform_precision.append(data['precision'][index])
+                uniform_recall.append(data['recall'][index])
+                uniform_queue_rate.append(data['queue_rate'][index])
+
+            uniform_precision_plots.append(uniform_precision)
+            uniform_recall_plots.append(uniform_recall)
+            uniform_queue_rate_plots.append(uniform_queue_rate)
+
+        lower_precision, median_precision, upper_precision = mstats.mquantiles(
+            uniform_precision_plots, prob=self.quantiles, axis=0)
+        lower_recall, median_recall, upper_recall = mstats.mquantiles(
+            uniform_recall_plots, prob=self.quantiles, axis=0)
+        lower_queue_rate, median_queue_rate, upper_queue_rate = mstats.mquantiles(
+            uniform_queue_rate_plots, prob=self.quantiles, axis=0)
```

Comment:
My feeling is that most the above can be moved to its own function, since none of this data relies on the axes or drawing objects. Refactoring this might make it easier to debug. Also; would the user be interested in this data? If so, they could access that function directly without drawing or we could store this data after `fit()` rather than the dictionaries in plot data -- depends on what you think would be more valuable.